dask/dataframe/rolling.py

```diff
@@ -57,3 +57,99 @@ def rolling(arg, window, *args, **kwargs):
 rolling_quantile = wrap_rolling(pd.rolling_quantile)
 rolling_apply = wrap_rolling(pd.rolling_apply)
 rolling_window = wrap_rolling(pd.rolling_window)
+
+def call_pandas_rolling_method_single(this_partition, rolling_kwargs,
+        method_name, method_args, method_kwargs):
+    # used for the start of the df/series
+    if this_partition.shape[0] < rolling_kwargs['window']:
+        raise NotImplementedError("Window larger than partition size")
+
+    method = getattr(this_partition.rolling(**rolling_kwargs), method_name)
+    return method(*method_args, **method_kwargs)
+
+def call_pandas_rolling_method_with_neighbor(prev_partition, this_partition,
+        rolling_kwargs, method_name, method_args, method_kwargs):
+    # used for everything except for the start
+
+    window = rolling_kwargs['window']
+    if prev_partition.shape[0] < window:
+        raise NotImplementedError("Window larger than partition size")
+
+    if window > 1:
+        extra = window - 1
+        combined = pd.concat([prev_partition.iloc[-extra:], this_partition])
+
+        method = getattr(combined.rolling(window), method_name)
+        applied = method(*method_args, **method_kwargs)
+        return applied.iloc[extra:]
+    else:
+        method = getattr(this_partition.rolling(window), method_name)
+        return method(*method_args, **method_kwargs)
+
+class Rolling(object):
+    # What you get when you do ddf.rolling(...) or similar
+    """Provides rolling window calculcations.
+
+    """
+
+    def __init__(self, obj, kwargs):
+        self.obj = obj # dataframe or series
+        self.rolling_kwargs = kwargs
+
+    def _call_method(self, method_name, *args, **kwargs):
+        args = list(args) # make sure dask does not mistake this for a task
+
+        old_name = self.obj._name
+        new_name = 'rolling-' + tokenize(
+            self.obj, self.rolling_kwargs, method_name, args, kwargs)
+
+        dsk = {(new_name, 0): (
+            call_pandas_rolling_method_single, (old_name, 0),
+            self.rolling_kwargs, method_name, args, kwargs)}
+        for i in range(1, self.obj.npartitions + 1):
+            dsk[new_name, i] = (
+                call_pandas_rolling_method_with_neighbor,
+                (old_name, i-1), (old_name, i),
+                self.rolling_kwargs, method_name, args, kwargs)
+
+        return self.obj._constructor(
+            merge(self.obj.dask, dsk),
+            new_name,
+            self.obj,
```

Comment:
Thanks, I'd seen the `_pd` attribute and not known what it was about. I just played monkey-see-monkey-do with constructing the graph there.

Oddly, even adding integer columns the tests still pass, but I can look at updating this appropriately.